src/Google/RequestReviewStatuses.php

```diff
@@ -11,79 +11,96 @@ class RequestReviewStatuses implements Service {
+
 
 	public const MC_ACCOUNT_REVIEW_LIFETIME = MINUTE_IN_SECONDS * 20; // 20 minutes
 
 	/**
-	 * Merges the different program statuses based on priority
-	 *
-	 * @param array $response
+	 * Merges the different program statuses, issues and cooldown period date.
 	 *
-	 * @return array
+	 * @param array $response Associative array containing the response data from Google API
+	 * @return array The computed status, with the issues and cooldown period.
 	 */
 	public function get_statuses_from_response( array $response ) {
 		$issues   = [];
 		$cooldown = 0;
-		$status   = 'APPROVED';
-
-		foreach ( $response as $program_type ) {
-			foreach ( $program_type['data']['regionStatuses'] as $region_status ) {
-
-				if (
-					isset( $region_status['reviewIneligibilityReasonDetails'] ) &&
-					isset( $region_status['reviewIneligibilityReasonDetails']['cooldownTime'] )
-				) {
-					$region_cooldown = intval( strtotime( $region_status['reviewIneligibilityReasonDetails']['cooldownTime'] ) );
-
-					if ( ! $cooldown || $region_cooldown > $cooldown ) {
-						$cooldown = $region_cooldown;
-					}
-				}
-
-				if ( $region_status['eligibilityStatus'] === self::DISAPPROVED || $region_status['eligibilityStatus'] === self::WARNING ) {
-					if ( $status !== self::DISAPPROVED ) {
-						$status = $region_status['eligibilityStatus'];
-					}
+		$status   = null;
 
-					$issues = array_merge( $issues, $region_status['reviewIssues'] ?? [] );
-				}
+		$valid_program_states = [ 'ENABLED', self::NO_OFFERS ];
 
-				if ( $status === self::DISAPPROVED ) {
-					continue;
-				}
-
-				if ( $region_status['eligibilityStatus'] === self::UNDER_REVIEW ) {
-					$status = self::UNDER_REVIEW;
-				}
-
-				if ( $status === self::UNDER_REVIEW ) {
-					continue;
-				}
+		foreach ( $response as $program_type ) {
 
-				if ( $region_status['eligibilityStatus'] === self::PENDING_REVIEW ) {
-					$status = self::PENDING_REVIEW;
-				}
+			// In case any Program is with no offers we consider it Onboarding
+			if ( $program_type['data']['globalState'] === self::NO_OFFERS ) {
+				$status = self::ONBOARDING;
+				break;
+			}
 
-				if ( $status === self::PENDING_REVIEW ) {
-					continue;
-				}
+			// In case any Program is not enabled or there are no regionStatuses we return null status
+			if ( ! isset( $program_type['data']['regionStatuses'] ) || ! in_array( $program_type['data']['globalState'], $valid_program_states, true ) ) {
+				$status = null;
+				break;
+			}
 
-				if ( $region_status['eligibilityStatus'] === self::ONBOARDING ) {
-					$status = self::ONBOARDING;
-				}
+			// otherwise we compute the new status, issues and cooldown period
+			foreach ( $program_type['data']['regionStatuses'] as $region_status ) {
+				$issues = array_merge( $issues, $region_status['reviewIssues'] ?? [] );
+				self::maybe_update_cooldown_period( $region_status, $cooldown );
+				self::maybe_update_status( $region_status['eligibilityStatus'], $status );
```

Comment:
As far as I know self could be used as well to refer to class scope based methods. Not necessarily static.

Reply:
Yeah, but it is confusing, if you check the codebase usually the self is used for static methods or for example if you have a parent class with the same method, so it is useful to use the self:: or parent:: to the tell which method you want to call but in this case, it does not seem necessary to use self. 

Some examples: 

https://github.com/woocommerce/google-listings-and-ads/blob/abe75172c58526ba326e56ee8fb94ebf63fed1ce/src/Product/Attributes/AttributeManager.php#L220

https://github.com/woocommerce/google-listings-and-ads/blob/de511f904ba86d20584af209ec5d039c48c676f6/src/Shipping/ShippingZone.php#L192

https://github.com/woocommerce/google-listings-and-ads/blob/de511f904ba86d20584af209ec5d039c48c676f6/src/Shipping/ShippingZone.php#L234

Reply:
For me doesn't seem confusing to use self for access class methods. But I changed it to make it less confusing since I don't really have a strong preference over it.